src/main/java/carpet/script/api/Auxiliary.java

```diff
@@ -453,6 +453,30 @@ else if (!interactable && targetBlock == null)
             return (_c, _t) -> finalRes; // pass through for variables
         });
 
+        expression.addLazyFunction("action_bar",2, (c, t, lv)->{
+            Value pVal = lv.get(0).evalValue(c);
+
+            PlayerEntity p;
+            Text message = Text.of(lv.get(1).evalValue(c).getString());
+
+            if(pVal instanceof ListValue){
+                List<Value> players = ((ListValue) pVal).getItems();
+
+                for (Value player:players) {
+                    if(!(player instanceof EntityValue && ((EntityValue)pVal).getEntity() instanceof PlayerEntity))
+                        throw new InternalExpressionException("'action_bar' only takes players as arguments");
+                    p =(PlayerEntity) ((EntityValue)player).getEntity();
+                    p.sendMessage(message,true);
+                }
+            }   else if(pVal instanceof EntityValue && ((EntityValue)pVal).getEntity() instanceof PlayerEntity){
+                p =(PlayerEntity) ((EntityValue)pVal).getEntity();
+                p.sendMessage(message,true);
+            }   else
+                throw new InternalExpressionException("'action_bar' requires a player as first argument");
+            
+            return (_c, _t) -> Value.NULL;
+        });
+
```

Comment:
Guess what? Style again. Also in other functions there are no empty newlines between things, and IEE should also suggest a list.
```suggestion
        expression.addLazyFunction("action_bar", 2, (c, t, lv) ->
        {
            Value pVal = lv.get(0).evalValue(c);
            PlayerEntity p;
            Text message = Text.of(lv.get(1).evalValue(c).getString());
            if (pVal instanceof ListValue)
            {
                List<Value> players = ((ListValue) pVal).getItems();
                for (Value player:players)
                {
                    if (!(player instanceof EntityValue && ((EntityValue) pVal).getEntity() instanceof PlayerEntity))
                        throw new InternalExpressionException("'action_bar' only takes players as arguments");
                    p = (PlayerEntity) ((EntityValue)player).getEntity();
                    p.sendMessage(message, true);
                }
            }
            else if (pVal instanceof EntityValue && ((EntityValue) pVal).getEntity() instanceof PlayerEntity)
            {
                p = (PlayerEntity) ((EntityValue) pVal).getEntity();
                p.sendMessage(message, true);
            }
            else
                throw new InternalExpressionException("'action_bar' requires a player or a player list as first argument");
            return (_c, _t) -> Value.NULL;
        });
```

Reply:
o ye forgot.